src/appsignal/heartbeat.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import logging
+from binascii import hexlify
+from contextlib import contextmanager
+from os import urandom
+from time import time
+from typing import Iterator, Literal, TypedDict
+
+from .client import Client
+from .config import Config
+from .transmitter import transmit
+
+
+EventKind = Literal["start"] | Literal["finish"]
+
+
+class Event(TypedDict):
+    name: str
+    id: str
+    kind: EventKind
+    timestamp: int
+
+
+class Heartbeat:
+    name: str
+    id: str
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.id = hexlify(urandom(8)).decode("utf-8")
+
+    def _event(self, kind: EventKind) -> Event:
+        return Event(name=self.name, id=self.id, kind=kind, timestamp=int(time()))
+
+    def _transmit(self, event: Event) -> None:
+        config = Client.config() or Config()
+        logger = logging.getLogger("appsignal")
+
+        if not config.is_active():
+            logger.debug("AppSignal not active, not transmitting heartbeat event")
+            return
+
+        url = f"{config.option('logging_endpoint')}/heartbeats/json"
+        try:
+            response = transmit(url, json=event)
+            if 200 >= response.status_code >= 299:
```

Comment:
```suggestion
            if 200 >= response.status_code <= 299:
```
It should be lower than 300, right?

Reply:
This is Python's somewhat strange, but intuitive once you're aware of it, chain comparison syntax! https://note.nkmk.me/en/python-chain-comparisons/

`200 >= response.status_code >= 299` means the same as `200 >= response.status_code && response.status_code >= 299`.

Reply:
Oh, wait, no, you're completely right, that is not what's implemented. It should be `200 <= response.status_code <= 299`.